2-Draft-Accepted/RFC00NN-PowerShellCore-Installer-Module.md

```diff
@@ -0,0 +1,78 @@
+
+## Motivation
+
+Installation and configuration of PowerShell Core on Windows is tricky due to the wide range of Windows versions that we support (7-10 and Server 2008R2-2016) and because of the usage of some "system" concepts shared by Windows PowerShell like the WinRM remoting plugin.
+
+## Goals
+
+* I can install PowerShell Core across all supported versions of Windows and Windows PowerShell *at scale*.
+* I can install PowerShell Core on certain versions of Windows and Windows PowerShell without remembering highly arcane commands or manually downloading a package from a well-known location.
+* I can uninstall PowerShell Core using Windows PowerShell
+* I can register PowerShell Core as a WinRM remoting endpoint across all supported versions of Windows and Windows PowerShell *at scale*.
+
+## Specification
+
+Ship a module on Gallery called `PowerShellCore` which contains the following cmdlets:
+
+* Install-PowerShellCore
+  * Download and install the latest PowerShell Core MSI and install it using the MSI PackageManagement provider
+* Update-PowerShellCore
+  * If PowerShell Core is already installed, this will grab the latest PowerShell Core MSI and do an in-place upgrade of the MSI-installed instance on the machine.
+* Uninstall-PowerShellCore
+  * Uninstall the PowerShell Core MSI using the MSI PackageManagement provider.
+* Register-PowerShellCoreEndpoint
+  * Call the Install-PowerShellRemoting.ps1 script in PowerShell Core's `$PSHome`.
+
+These cmdlets allow me to do things like:
+
+```powershell
+Invoke-Command -ComputerName $TargetNode -Credential DOMAIN\BestAdmin -ScriptBlock {Install-Module PowerShellCore -Force; Install-PowerShellCore; Register-PowerShellCoreEndpoint}
+```
+
+`PowerShellCore`should start as a 0.1 release so that we can make breaking changes as we gather feedback in PowerShell Core 6.0 beta.
+
+Right now, I think it's perfectly okay to ship all of these with nothing but common parameters.
+Based on feedback to this RFC, we should prioritize a set of parameters that might be useful.
+(For example, we may eventually add a `-Source` to `Install/Upgrade-PowerShellCore` that points to `GitHub` and some other, more official, source.)
+
+## Alternate Proposals and Considerations
+
+### Nano Server
+
+Given that Nano Server currently has Docker containers with the latest version of PowerShell Core, Nano Server is out of scope for this RFC.
+However, it's perfectly reasonable that a future version of `PowerShellCore` could extend itself to support Nano Server via the same set of cmdlets.
+
+### Servicing
+
+MSIs have well-known patterns for updates and servicing (e.g. WSUS, SCCM, etc).
+While it's curently unclear *where* we should publish these MSIs to best enable these servicing patterns,
+delivering an out-of-band application via an MSI is better than other alternatives for notifying administrators and user of security fixes and critical updates.
+
+### Tracking "Portable" Copies of PowerShell
```

Comment:
The term "Portable" seems fine to me. It's widely used and well-understood, in my experience.